scrapy/resolver.py

```diff
@@ -4,17 +4,25 @@
 from scrapy.utils.datatypes import LocalCache
 
 # TODO: cache misses
-# TODO: make cache size a setting
 
 dnscache = LocalCache(10000)
 
 class CachingThreadedResolver(ThreadedResolver):
+    def __init__(self, reactor, settings):
```

Comment:
But it could get inconvenient if there are yet more settings in the future, such as a full resolver based on `twisted.names` could provide (upstream caching resolver IPs, use EDNS, do DNSSEC validation, IPv4/v6 preference and what not).